zebra-state/src/service/non_finalized_state/chain.rs

```diff
@@ -509,6 +540,10 @@ impl UpdateWith<Option<orchard::ShieldedData>> for Chain {
         orchard_shielded_data: &Option<orchard::ShieldedData>,
     ) -> Result<(), ValidateContextError> {
         if let Some(orchard_shielded_data) = orchard_shielded_data {
+            for cm_x in orchard_shielded_data.note_commitments() {
```

Comment:
Missing docs: let's explain why we can't remove the orchard notes in `revert_chain_state_with`

Reply:
Added in dce91134